src/objects/Target.gd

```diff
@@ -12,6 +12,12 @@ func on_raycast_hit(player_id: int) -> void:
 
 
 remote func destroy_self(player_id: int) -> void:
-	emit_signal("target_destroyed", player_id)
-	get_parent().remove_child(self)
 	queue_free()
+	emit_signal("target_destroyed", player_id)
+
+
+func _on_Target_body_entered(body: Node) -> void:
+	if body.is_in_group("Arrow"):
+		if get_tree().has_network_peer():
```

Comment:
`if get_tree().has_network_peer() and get_tree().is_network_server()` (I think)

This makes it so only the server can decide that a target has been destroyed. Otherwise, each client might try and call this for each target